tools/workflow/cleaner/lambda/clean.go

```diff
@@ -79,3 +154,19 @@ func shouldDelete(lf *lambda.FunctionConfiguration) (bool, error) {
 	}
 	return false, nil
 }
+
+func shouldDeleteLayer(layerList *lambda.LayersListItem) *string {
```

Comment:
I am currently using this function to check provide a LayerARN to delete.

Reply:
Right, and I'm suggesting that instead of your current implementation use return a bool instead. Keeping the same name and returning a bool may make more sense than my original suggestion. You will be able to clean up the code a bit at the calling site by just doing something like

```
if (shouldDeleteLayer(layerList *lambda.LayersListItem))
```

The only thing is, if you are going to keep the `shouldDeleteLayer` name you should probably also do more than just check the regex. You should also check the creation time to see if it's expired. Otherwise, the function name doesn't make sense, because more logic is required than just the regex to see if  the `layer should be deleted`.

Reply:
[I added a comment](https://github.com/aws-observability/aws-otel-collector/pull/2611/files#r1476909825) to expand on how the caller logic could be simplified by enhancing to `shouldDeleteLayer` func to perform all the decision making.

Reply:
Ahh Yeah just saw, i got your suggestion.

Reply:
Commited the changes to let shouldDeleteLayer do more whether to delete